<?xml version="1.0"?>
<launch>
  
  <!-- Determine these using rosrun pointgrey_camera_driver list_cameras -->
  <arg name="camera_serial_left" />
  <arg name="camera_serial_right" />
  <arg name="conf_file_left"  default="$(find opt_calibration)/conf/pg_default.yaml" />
  <arg name="conf_file_right" default="$(find opt_calibration)/conf/pg_default.yaml" />
  <arg name="stereo_name"     default="stereo_pg"/>
  <arg name="calibrated"      default="true" />

  <!-- Left Camera -->
  <group ns="$(arg stereo_name)/left">
  
    <node pkg="nodelet" type="nodelet" name="camera_nodelet_manager_left" args="manager" />

    <node pkg="nodelet" type="nodelet" name="camera_nodelet_left"
          args="load pointgrey_camera_driver/PointGreyCameraNodelet camera_nodelet_manager_left" >
          
      <param name="frame_id"   value="/$(arg stereo_name)" />
      <param name="serial"     value="$(arg camera_serial_left)" />
      <rosparam command="load" file="$(arg conf_file_left)" />
      
      <!-- Use the camera_calibration package to create this file -->
      <param name="camera_info_url" if="$(arg calibrated)"
             value="file://$(find opt_calibration)/conf/stereo_calib_left.yml" />
    </node>

    <node pkg="nodelet" type="nodelet" name="image_proc_debayer_left"
          args="load image_proc/debayer camera_nodelet_manager">
      <remap from="image_raw"   to="left/image_raw"/>
      <remap from="image_mono"  to="left/image_mono"/>
      <remap from="image_color" to="left/image_color"/>
    </node>
    
  </group>

  <!-- Right Camera -->
  <group ns="$(arg stereo_name)/right">
   
    <node pkg="nodelet" type="nodelet" name="camera_nodelet_manager_right" args="manager" />

    <node pkg="nodelet" type="nodelet" name="camera_nodelet_right"
          args="load pointgrey_camera_driver/PointGreyCameraNodelet camera_nodelet_manager_right" >
          
      <param name="frame_id"   value="/$(arg stereo_name)" />
      <param name="serial"     value="$(arg camera_serial_right)" />
      <rosparam command="load" file="$(arg conf_file_right)" />
      
      <!-- Use the camera_calibration package to create this file -->
      <param name="camera_info_url" if="$(arg calibrated)"
             value="file://$(find opt_calibration)/conf/stereo_calib_right.yml" />
    </node>

    <node pkg="nodelet" type="nodelet" name="image_proc_debayer_right"
          args="load image_proc/debayer camera_nodelet_manager">
      <remap from="image_raw"   to="right/image_raw"/>
      <remap from="image_mono"  to="right/image_mono"/>
      <remap from="image_color" to="right/image_color"/>
    </node>
    
  </group>
  
</launch>
